src/server/user-controller.spec.js

```diff
@@ -0,0 +1,136 @@
+import sinon from 'sinon';
+import events from 'events';
+import httpMocks from 'node-mocks-http';
+import sut from './user-request-handlers';
+import { SAVE_USER_RESULT } from '../domain/user-management';
+
+describe('user-request-handlers', () => {
+
+  const sandbox = sinon.createSandbox();
+  let clientDataMapper;
+  let userService;
+  let mailService;
+
+  beforeEach(() => {
+    clientDataMapper = {
+      dbUserToClientUser: sandbox.stub()
+    };
+    userService = {
+      createUser: sandbox.stub()
+    };
+    mailService = {
+      sendRegistrationVerificationLink: sandbox.stub()
+    };
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe('handleUsersPost', () => {
+    let req;
+    let res;
+    const mappedUser = {};
+
+    describe('with all data correctly provided', () => {
+      beforeEach(done => {
+        req = httpMocks.createRequest({
+          protocol: 'https',
+          headers: { host: 'localhost' },
+          body: { username: 'test1234', email: 'test@test.com', password: 'abcd1234' }
+        });
+        res = httpMocks.createResponse({ eventEmitter: events.EventEmitter });
+
+        res.on('end', done);
+
+        userService.createUser.resolves({ result: SAVE_USER_RESULT.success, user: { verificationCode: 'verificationCode' } });
+        clientDataMapper.dbUserToClientUser.returns(mappedUser);
+
+        sut.handlePostUser({ req, res, clientDataMapper, mailService, userService });
+      });
+
+      it('should set the status code on the response to 200', () => {
+        expect(res.statusCode).toBe(200);
```

Comment:
Shouldn't this be a 201 on POST? (I know this is probably old stuff)

Reply:
See other comment :)